reflex/components/sonner/toast.py

```diff
@@ -0,0 +1,267 @@
+
+        Returns:
+            The object as a json string.
+        """
+        from reflex.utils.serializers import serialize
+
+        return self.__config__.json_dumps(
+            {format.to_camel_case(key): value for key, value in self.dict().items()},
+            default=serialize,
+        )
+
+
+class ToastProps(PropsBase):
+    """Props for the toast component."""
+
+    # Toast's description, renders underneath the title.
+    description: str = ""
+
+    # Whether to show the close button.
+    close_button: bool = False
+
+    # Dark toast in light mode and vice versa.
+    invert: bool = False
+
+    # Control the sensitivity of the toast for screen readers
+    important: bool = False
+
+    # Time in milliseconds that should elapse before automatically closing the toast.
+    duration: int = 4000
+
+    # Position of the toast.
+    position: LiteralPosition = "bottom-right"
+
+    # If false, it'll prevent the user from dismissing the toast.
+    dismissible: bool = True
+
+    # TODO: fix serialization of icons for toast? (might not be possible yet)
+    # Icon displayed in front of toast's text, aligned vertically.
+    # icon: Optional[Icon] = None
+
+    # TODO: fix implementation for action / cancel buttons
+    # Renders a primary button, clicking it will close the toast.
+    # action: str = ""
+
+    # Renders a secondary button, clicking it will close the toast.
+    # cancel: str = ""
+
+    # Custom id for the toast.
+    id: str = ""
+
+    # Removes the default styling, which allows for easier customization.
+    unstyled: bool = False
+
+    # Custom style for the toast.
+    style: Style = Style()
+
+    # Custom style for the toast primary button.
+    # action_button_styles: Style = Style()
+
+    # Custom style for the toast secondary button.
+    # cancel_button_styles: Style = Style()
+
+
+class Toaster(Component):
+    """A Toaster Component for displaying toast notifications."""
+
+    library = "sonner@1.4.41"
+
+    tag = "Toaster"
+
+    # the theme of the toast
+    theme: Var[str] = color_mode
+
+    # whether to show rich colors
+    rich_colors: Var[bool] = Var.create_safe(True)
+
+    # whether to expand the toast
+    expand: Var[bool] = Var.create_safe(True)
+
+    # the number of toasts that are currently visible
+    visible_toasts: Var[int]
+
+    # the position of the toast
+    position: Var[LiteralPosition] = Var.create_safe("bottom-right")
+
+    # whether to show the close button
+    close_button: Var[bool] = Var.create_safe(False)
+
+    # offset of the toast
+    offset: Var[str]
+
+    # directionality of the toast (default: ltr)
+    dir: Var[str]
+
+    # Keyboard shortcut that will move focus to the toaster area.
+    hotkey: Var[str]
+
+    # Dark toasts in light mode and vice versa.
+    invert: Var[bool]
+
+    # These will act as default options for all toasts. See toast() for all available options.
+    toast_options: Var[ToastProps]
```

Comment:
When i set defaults here, they do not seem to be respected.

I tried setting it like

```python
rx.toast.provider(toast_options=rx.toast.options(close_button=True, duration=300), close_button=True)
```

And the toasts still use the default 4000ms duration and do not have close buttons, unless i pass no options at all.

For example, i'm trying to toast like this

```python
                    yield rx.toast.warning(
                        f"Removed {item.text} from {list_name}",
                        description="Whoops, this was irreversible"
                    )
```

It seems that any options passed myself overwrite all of the toaster options with the original defaults.

Reply:
so i updated ToastProps to make all of the fields optional, which seemed to help; but now i'm finding that some of the fields in `ToastProps` don't take effect when passed to the `toast_options` prop.

- description
- position
- close_button
(maybe others, didn't try them all)